tests/unit/codegen/test12.cpp

```diff
@@ -10,5 +10,5 @@ void webMain()
 {
 	const char* str="-0.832050323486328125";
 	double ret=strtod(str, NULL);
-	assertEqual(ret,-0.832050323486328125,"High precision strtod");
+	assertEqual<double>(ret,(double)-0.832050323486328125,"High precision strtod");
```

Comment:
Literals are double by default. What is the problem supposed to be?

Reply:
I will check again, but basically the problem was that the float specialization of assertEqual was taken:
```
template<class T>
void assertEqual(const T value, const T expected, const char* msg)
{
        assertEqualImpl(value==expected, msg);
}
```

Reply:
It cannot make sense, at the very least `ret` is certainly double. So T would be double. But anyway the literal is certain double as well.